### PR TITLE
fix: make token retrieval in _get_auth asynchronous to prevent blocki…

### PR DESCRIPTION
**Pull request by nimanikoo**

### CODE REVIEW

**Comment by Swolehokage** [comment]
@CodiumAI-Agent /describe
--pr_description.publish_labels=false
This pull request includes changes to the `python/rtclient/__init__.py` file to enhance the authentication process by making it asynchronous. The most important changes are:

Authentication process improvements:

* Modified the `_get_auth` method to be asynchronous by adding the `async` keyword and using `await` for the `get_token` call. (`python/rtclient/__init__.py`)
* Updated the `connect` method to await the `_get_auth` method, ensuring it handles asynchronous authentication properly. (`python/rtclient/__init__.py`)

File: python/rtclient/__init__.py [hunk 1]
```diff
@@ -138,10 +138,10 @@ def __init__(
     def _user_agent(self):
         return "ms-rtclient-0.4.3"
 
-    def _get_auth(self):
+    async def _get_auth(self):
         if self._token_credential:
             scopes = ["https://cognitiveservices.azure.com/.default"]
-            token = self._token_credential.get_token(scopes)
+            token = await self._token_credential.get_token(scopes)
```
Comment (iMicknl) [inline python/rtclient/__init__.py:144]: Should you not leverage the async credentials as well, which you need to import from [azure.identity.aio](https://aka.ms/azsdk/python/identity/aio/docs)? Not sure if the current proposal will work in a non blocking way.

See https://learn.microsoft.com/en-us/python/api/overview/azure/identity-readme?view=azure-python#async-credentials

**Comment by glecaros** [comment]
Hi @nimanikoo, thank you for the PR, I've taken the liberty to update it to use the `AsyncTokenCredential` and updated the usage accordingly. Thanks again!

**Comment by nimanikoo** [comment]
Hi @glecaros 

Thank you so much for merging my contribution. I hope it helps the project, and I look forward to contributing more in the future.

**Pull request merged**